src/commands/upload_file.ts

```diff
@@ -169,66 +169,29 @@ new CLICommand({
 				shouldBundle
 			});
 
-			const results = await Promise.all(
-				filesToUpload.map(async (fileToUpload) => {
-					const {
-						parentFolderId,
-						wrappedEntity,
-						destinationFileName,
-						drivePassword,
-						driveKey: fileDriveKey
-					} = fileToUpload;
-
-					return await (async () => {
-						if (await parameters.getIsPrivate()) {
-							const driveId = await arDrive.getDriveIdForFolderId(parentFolderId);
-							const driveKey =
-								fileDriveKey ??
-								(await parameters.getDriveKey({ driveId, drivePassword, useCache: true }));
-
-							if (isFolder(wrappedEntity)) {
-								return arDrive.createPrivateFolderAndUploadChildren({
-									parentFolderId,
-									wrappedFolder: wrappedEntity,
-									driveKey,
-									destParentFolderName: destinationFileName,
-									conflictResolution,
-									prompts: folderUploadConflictPrompts
-								});
-							} else {
-								return arDrive.uploadPrivateFile({
-									parentFolderId,
-									wrappedFile: wrappedEntity,
-									driveKey,
-									destinationFileName,
-									conflictResolution,
-									prompts: fileUploadConflictPrompts
-								});
-							}
-						} else {
-							if (isFolder(wrappedEntity)) {
-								return arDrive.createPublicFolderAndUploadChildren({
-									parentFolderId,
-									wrappedFolder: wrappedEntity,
-									destParentFolderName: destinationFileName,
-									conflictResolution,
-									prompts: folderUploadConflictPrompts
-								});
-							} else {
-								return arDrive.uploadPublicFile({
-									parentFolderId,
-									wrappedFile: wrappedEntity,
-									destinationFileName,
-									conflictResolution,
-									prompts: fileUploadConflictPrompts
-								});
-							}
-						}
-					})();
-				})
+			const uploadStats: ArDriveUploadStats[] = await Promise.all(
+				filesToUpload.map(
+					async ({ parentFolderId, wrappedEntity, destinationFileName, driveKey, drivePassword }) => {
+						driveKey ??= (await parameters.getIsPrivate())
+							? await parameters.getDriveKey({
+									driveId: await arDrive.getDriveIdForFolderId(parentFolderId),
+									drivePassword,
+									useCache: true
+							  })
+							: undefined;
+
+						return { wrappedEntity, driveKey, destFolderId: parentFolderId, destName: destinationFileName };
+					}
+				)
 			);
 
-			console.log(JSON.stringify(formatResults(results), null, 4));
+			const results = await arDrive.uploadAllEntities({
+				entitiesToUpload: uploadStats,
+				conflictResolution,
+				prompts: folderUploadConflictPrompts
```

Comment:
Nit: This name always throws me off. `fileOrFolderUploadConflictPrompts` would likely make more sense since they're composed together.

Reply:
Changed that one over to `fileAndFolderUploadConflictPrompt`, since they contain both: 6c14939